sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -31,6 +31,57 @@
 from ..utils import longest_equal_range
 
 
+def transfer_if_needed(*tensors: Tuple[ShardedTensor]) -> List[ShardedTensor]:
```

Comment:
I would like to put this in \_\_init\_\_ at well since that is not the only place it's being used. But the top level type hints for this functions will make the imports messy.

Reply:
We want to keep in sharding as all of these wrapping behavior should be sharded specific.